### DISCUSSION TITLE
Could not get RNS status (with rnstatus)

### DISCUSSION THREAD

**Discussion by resiliencetheatre** [markqvist/Reticulum#533] [Help & Questions]
I have embedded entity where I use rnsd and it works well (with meshchat). 

I can run successfully rnsd:

`[2024-08-26 00:02:38] [Notice] Started rnsd version 0.7.5`

But rnstatus returns error when I run it:

```
# rnstatus -v
Could not get RNS status
```

And rnsd verbose log outputs:

```
[2024-08-26 00:02:45] [Error] An error ocurred while handling RPC call from local client: digest received was wrong
```

Announces and meshchat works and messages gets delivered just fine. What could be a problem here?

**Comment by resiliencetheatre** [comment]
It might be that running 'rnstatus' with wrong config path generated this error. Plain rnstatus creates ~/.reticulum but with correct config path it seems to work. User error.

```
# rnstatus --config=/opt/meshchat

 Shared Instance[37428]
    Status    : Up
    Serving   : 1 program
    Rate      : 1.00 Gbps
    Traffic   : 10.71 KB↑
                4.12 KB↓

 RNodeInterface[RNode LoRa Interface]
    Status    : Up
    Mode      : Full
    Rate      : 3.12 kbps
    Airtime   : 0.0% (15s), 0.11% (1h)
    Ch.Load   : 0.0% (15s), 1.03% (1h)
    Traffic   : 3.48 KB↑
                12.79 KB↓

```

**Reply by faragher** [reply]
Good to know! ~/.reticulum/config is for Reticulum, but apparently Meshchat uses a different config path. This could be the source of some of the issues I've seen lately. Thanks!

**Reply by markqvist** [reply]
Hmm, I think that in most cases, MeshChat should really use the default Reticulum configuration available on the system, if there is one. It may be a good idea to change this @liamcottle. There's obviously situations where you'd want to use another config path, but maybe there could just be a command-line parameter to change this, like Sideband and NomadNet does.

**Reply by liamcottle** [reply]
@markqvist MeshChat does indeed use `~/.reticulum` for reticulum config by default.

In this case, @resiliencetheatre had explicitly set a custom path of `/opt/meshchat` in their systemd service file, hence why this happened :)

https://github.com/resiliencetheatre/rpi4edgemapdisplay/blob/abe5d9b1d07b603717f861d091114aecba84c3d5/board/raspberrypi/fs_edgemap/etc/systemd/system/meshchat.service#L8

**Reply by markqvist** [reply]
Ah, that makes sense! I was also kinda wondering why I'd never experienced any trouble. Thanks for clarifying!
